java/src/main/java/org/hyperledger/fabric/client/GatewayImpl.java

```diff
@@ -148,14 +148,20 @@ public Network getNetwork(final String networkName) {
 
     @Override
     public Proposal newSignedProposal(final byte[] bytes, final byte[] signature) {
+        ProposalImpl result = (ProposalImpl) newProposal(bytes);
+        result.setSignature(signature);
+        return result;
+    }
+
+    @Override
+    public Proposal newProposal(final byte[] bytes) {
```

Comment:
Avoid the casts by returning the implementation type from the `newType()` methods. The client application works with the interface, which declares the interface type so this should be hidden from application code but avoids an unchecked cast.
```suggestion
        ProposalImpl result = newProposal(bytes);
        result.setSignature(signature);
        return result;
    }

    @Override
    public ProposalImpl newProposal(final byte[] bytes) {
```